internal/clitrigger/websocket_trigger.go

```diff
@@ -0,0 +1,114 @@
+package clitrigger
+
+import (
+	"context"
+	"encoding/json"
+	"time"
+
+	"github.com/hashicorp/go-hclog"
+	"nhooyr.io/websocket"
+
+	"github.com/raito-io/cli/internal/auth"
+	"github.com/raito-io/cli/internal/target"
+)
+
+type WebsocketCliTrigger struct {
+	websocketUrl string
+}
+
+func NewWebsocketCliTrigger(websocketUrl string) *WebsocketCliTrigger {
+	return &WebsocketCliTrigger{
+		websocketUrl: websocketUrl,
+	}
+}
+
+func (s *WebsocketCliTrigger) TriggerChannel(ctx context.Context, config *target.BaseConfig) (chan TriggerEvent, error) {
+	options := websocket.DialOptions{
+		HTTPHeader: map[string][]string{},
+	}
+
+	err := auth.AddTokenToHeader(&options.HTTPHeader, config)
+	if err != nil {
+		return nil, err
+	}
+
+	c, _, err := websocket.Dial(ctx, s.websocketUrl, &options)
+	if err != nil {
+		return nil, err
+	}
+
+	heartbeat(ctx, c, config.BaseLogger)
+
+	return triggerChannel(ctx, c), nil
+}
+
+func heartbeat(ctx context.Context, c *websocket.Conn, logger hclog.Logger) {
+	go func() {
+		heartbeatMsg := []byte("{\"message\": \"heartbeat\"}")
+
+		ticker := time.NewTicker(time.Duration(9) * time.Minute)
```

Comment:
Why 9 minutes? If the timeout is 10 minutes, then maybe we should rather do the heartbeat faster to make sure that if there is a failure we get some more chances to try?

Reply:
ok